pkg/kubelet/kubelet.go

```diff
@@ -261,7 +261,7 @@ func getContainerEtcHostsPath() string {
 
 // SyncHandler is an interface implemented by Kubelet, for testability
 type SyncHandler interface {
-	HandlePodAdditions(pods []*v1.Pod)
+	HandlePodAdditions(ctx context.Context, pods []*v1.Pod)
```

Comment:
This is one of the reasons I thought about moving slowly and starting from the subsustems. Adding context to pod workers is something which I'd be very careful about, and on top of that adding only to pod additions seems weird.

I know adding to all functions is a ton of work, but still.

Reply:
Thank you for the review. I agree, adding ctx to only one method is not good. Removed. PTAL.

Reply:
I think this is a valid point, and I generally agree. However, I can't help but wonder - since there's a lot of work to be done in this area, it would be beneficial for us to define and document the criteria for selectively adding context. This would help us decide where it makes sense to include context and where it doesn't, and it would also enable us to better track pending items.